src/main/java/com/lucidworks/spark/SolrRelation.java

```diff
@@ -51,14 +61,23 @@ public SolrRelation(SQLContext sqlContext, Map<String,String> config, DataFrame
       throw new IllegalArgumentException("SQLContext cannot be null!");
 
     this.sqlContext = sqlContext;
-
+    this.sc =  new JavaSparkContext(sqlContext.sparkContext());
+    if (optionalParam(config, PRESERVE_SCHEMA, "N").equals("Y")) {
```

Comment:
I'd throw a Boolean.parseBoolean in there too, to allow for Y or True or true as input values